### PR TITLE
Proxito middleware: reset to original urlconf after request

### PR DESCRIPTION
**Pull request by stsewd**
This also fixes a flaky test.
We hit this before proxito too
https://github.com/readthedocs/readthedocs.org/blob/8ebded69160f495c56e2966da90bad3be2dc9f5f/readthedocs/core/middleware.py#L132-L136

### CODE REVIEW

**Comment by stsewd** [comment]
Ok... it doesn't fail on my computer anymore, but it still fails on travis :upside_down_face:

File: readthedocs/proxito/middleware.py [hunk 1]
```diff
@@ -199,6 +200,10 @@ def process_response(self, request, response):  # noqa
         * For custom domains, check the HSTS values on the Domain object.
           The domain object should be saved already in request.domain.
         """
+        # Reset URLconf for this thread
+        # to the original one.
```
Comment (humitos) [inline readthedocs/proxito/middleware.py:204]: why not put the comment in just one line? :smile:

**Comment by ericholscher** [comment]
@stsewd yea, the original test works for me locally, but fails on travis. Really not sure what's going on, but the request-based URLConf stuff is definitely buggy.

**Comment by stsewd** [comment]
@ericholscher tests should be passing now, to replicate locally I had to run the tests a lot of times till I got an error, then trying with/without my fix several times to see if failed or not...

File: readthedocs/proxito/middleware.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/proxito/middleware.py:204]: I just like short lines to have more space in my monitor :man_shrugging:

**Pull request merged**